src/ActivitylogServiceProvider.php

```diff
@@ -43,9 +44,10 @@ public function register()
 
     public static function determineActivityModel(): string
     {
-        $activityModel = config('activitylog.activity_model') ?? Activity::class;
+        $activityModel = config('activitylog.activity_model') ?? ActivityModel::class;
 
-        if (! is_a($activityModel, Activity::class, true)) {
+        if (! is_a($activityModel, Activity::class, true)
```

Comment:
The first one `is_a($activityModel, Activity::class, true)` checks for instance of the interface/contract - see Line#7.
The second is also right, if it's not a Model it couldn't use the save and all the other eloquent methods.

Reply:
In the original provider check if the model class was an instance of the `Spatie\Activitylog\Models\Activity` the check was done with the is_a function also in the original class.
I've add the  `|| ! is_a($activityModel, Model::class, true))` because now the requirements is also that the model has to extend `Illuminate\Database\Eloquent\Model`

Reply:
You're right, I was mistaken because of the names 🤐

Reply:
So I think that the check on `is_a($activityModel, Activity::class, true) ` (`Activity` being the interface) means this is a breaking change though.

Reply:
Yes you're right. 
For backwards compatibily it could be changed like this:
```php
if ((!is_a($activityModel, Activity::class, true) && !is_a($activityModel, ActivityModel::class, true))
            || ! is_a($activityModel, Model::class, true)) {
            throw InvalidConfiguration::modelIsNotValid($activityModel);
        }
```
So it can be an implementation of Activity Interface or extend the Activity Model shipped with the package. This should work for backword compatibility. What do you think about it?

Reply:
Sounds good. We can add an issue to review this in the next major version.

Reply:
Ok I could change this into my code and push the commits next week.

Reply:
But this will be the same.
The package `ActivityModel` implements the `ActivityContract` - so the `if` will fail/pass in 100% the same cases.
So I would say that it's not a BC and doesn't need adjustments in the condition.

```php
// current own model
class MyActivity extends ActivityModel { ... }
```
This is the only way to create a custom ActivityModel at the moment. So after a `composer update` this class will also implement the interface (because the base model does it).

Reply:
You're right, it would work indeed. All the better!